src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -190,9 +190,11 @@ function createInstanceTypeChecker(expectedClass) {
     if (!(props[propName] instanceof expectedClass)) {
       var locationName = ReactPropTypeLocationNames[location];
       var expectedClassName = expectedClass.name || ANONYMOUS;
+      var actualClassName = props[propName].constructor.name;
```

Comment:
Oh right, `null` doesn't actually come into this check because it does the `isRequired` check first.